mempool/tests/mod.rs

```diff
@@ -1225,6 +1239,125 @@ async fn mempool_update_pruned_account() {
     }
 }
 
+#[test(tokio::test(flavor = "multi_thread", worker_threads = 10))]
+async fn mempool_update_create_staker_twice() {
+    let time = Arc::new(OffsetTime::new());
+    let env = VolatileEnvironment::new(10).unwrap();
+
+    let key_pair = ed25519_key_pair(ACCOUNT_SECRET_KEY);
+    let address = Address::from_any_str(STAKER_ADDRESS).unwrap();
+
+    // This is the transaction produced in the block
+    let tx = TransactionBuilder::new_create_staker(
+        &key_pair,
+        &key_pair,
+        Some(address.clone()),
+        100_000_000.try_into().unwrap(),
+        100.try_into().unwrap(),
+        1,
+        NetworkId::UnitAlbatross,
+    )
+    .unwrap();
+
+    // This is a slightly different transaction, that also creates the same staker
+    let tx_dup = TransactionBuilder::new_create_staker(
+        &key_pair,
+        &key_pair,
+        Some(address),
+        100.try_into().unwrap(),
+        0.try_into().unwrap(),
+        1,
+        NetworkId::UnitAlbatross,
+    )
+    .unwrap();
+
+    let txns = vec![tx_dup];
+
+    let adopted_txns = vec![tx];
+
+    let mut adopted_micro_blocks = vec![];
+
+    adopted_micro_blocks.push((
+        Blake2bHash::default(),
+        create_dummy_micro_block(Some(adopted_txns.to_vec())),
+    ));
+
+    log::debug!("Done generating adopted micro block");
+
+    let blockchain = Arc::new(RwLock::new(
+        Blockchain::new(env, NetworkId::UnitAlbatross, time).unwrap(),
+    ));
+
+    // Create mempool and subscribe with a custom txn stream.
+    let mempool = Mempool::new(blockchain.clone(), MempoolConfig::default());
+    let mut hub = MockHub::new();
+    let mock_id = MockId::new(hub.new_address().into());
+    let mock_network = Arc::new(hub.new_network());
+
+    // Send txns to mempool
+    send_txn_to_mempool(&mempool, mock_network, mock_id, txns).await;
+
+    assert_eq!(
+        mempool.num_transactions(),
+        1,
+        "Number of txns in the mempools is not what is expected"
+    );
+
+    //We need a block producer to produce blocks
+    let producer = BlockProducer::new(signing_key(), voting_key());
+
+    {
+        let bc = blockchain.upgradable_read();
+
+        let block = producer.next_micro_block(
+            &bc,
+            bc.time.now(),
+            0,
+            None,
+            vec![],
+            adopted_txns,
+            vec![0x41],
+        );
+
+        assert_eq!(
+            Blockchain::push(bc, Block::Micro(block)),
+            Ok(PushResult::Extended)
+        );
+    }
+
+    // Call mempool update
+    mempool.mempool_update(&adopted_micro_blocks[..], &[].to_vec());
+
+    // Get txns from mempool
+    let updated_txns = mempool.get_transactions_for_block(10_000);
+
+    assert_eq!(
+        updated_txns.len(),
+        0,
+        "Number of txns is not what is expected"
+    );
+
+    {
```

Comment:
This extra scope could be removed since the end of the scope is the end of the function